nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/ml/textract/GetAwsTextractJobStatus.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.aws.ml.AwsMachineLearningJobStatusGetter;
+
+@Tags({"Amazon", "AWS", "ML", "Machine Learning", "Textract"})
+@CapabilityDescription("Retrieves the current status of an AWS Textract job.")
+@SeeAlso({StartAwsTextractJob.class})
+public class GetAwsTextractJobStatus extends AwsMachineLearningJobStatusGetter<AmazonTextractClient> {
+    public static final String DOCUMENT_ANALYSIS = "Document Analysis";
+    public static final String DOCUMENT_TEXT_DETECTION = "Document Text Detection";
+    public static final String EXPENSE_ANALYSIS = "Expense Analysis";
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("type-of-service")
+            .displayName("Type of textract")
+            .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+            .allowableValues(DOCUMENT_ANALYSIS, DOCUMENT_TEXT_DETECTION, EXPENSE_ANALYSIS)
+            .required(true)
+            .defaultValue("Document Analysis")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    private static final List<PropertyDescriptor> TEXTRACT_PROPERTIES = ImmutableList.<PropertyDescriptor>builder()
+            .addAll(PROPERTIES)
+            .add(TYPE)
+            .build();;
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return TEXTRACT_PROPERTIES;
+    }
+
+    @Override
+    protected AmazonTextractClient createClient(ProcessContext context, AWSCredentialsProvider credentialsProvider, ClientConfiguration config) {
+        return (AmazonTextractClient) AmazonTextractClient.builder()
+                .withRegion(context.getProperty(REGION).getValue())
+                .withCredentials(credentialsProvider)
+                .build();
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+        String typeOfTextract = context.getProperty(TYPE).evaluateAttributeExpressions().getValue();
+
+        String awsTaskId = flowFile.getAttribute(AWS_TASK_ID_PROPERTY);
+        JobStatus jobStatus = getTaskStatus(typeOfTextract, getClient(), awsTaskId);
+        if (JobStatus.SUCCEEDED == jobStatus) {
+            Object task = getTask(typeOfTextract, getClient(), awsTaskId);
+            writeToFlowFile(session, flowFile, task);
+            session.transfer(flowFile, REL_SUCCESS);
+        }
+
+        if (JobStatus.IN_PROGRESS == jobStatus) {
+            session.transfer(flowFile, REL_RUNNING);
+        }
+
+        if (JobStatus.PARTIAL_SUCCESS == jobStatus) {
+            session.transfer(flowFile, REL_THROTTLED);
+        }
+
+        if (JobStatus.FAILED == jobStatus) {
```

Comment:
It looks like these conditionals shoudl be chained together with `else if`

Reply:
Fixed, thx.